diff --git a/src/libsodium/sodium/runtime.c b/src/libsodium/sodium/runtime.c
index 78f0a58..1adde5c 100644
--- a/src/libsodium/sodium/runtime.c
+++ b/src/libsodium/sodium/runtime.c
@@ -65,7 +65,7 @@ static void
 _cpuid(unsigned int cpu_info[4U], const unsigned int cpu_info_type)
 {
 #if defined(_MSC_VER) && \
-    (defined(_M_X64) || defined(_M_AMD64) || defined(_M_IX86))
+    (defined(_M_X64) || defined(_M_AMD64) || defined(_M_IX86) || WINAPI_DESKTOP)
     __cpuid((int *) cpu_info, cpu_info_type);
 #elif defined(HAVE_CPUID)
     cpu_info[0] = cpu_info[1] = cpu_info[2] = cpu_info[3] = 0;
diff --git a/src/libsodium/sodium/utils.c b/src/libsodium/sodium/utils.c
index c9f11ba..9b8f684 100644
--- a/src/libsodium/sodium/utils.c
+++ b/src/libsodium/sodium/utils.c
@@ -10,13 +10,21 @@
 #include <stdlib.h>
 #include <string.h>
 
+#ifdef _WIN32
+# if !defined(WINAPI_FAMILY) || (WINAPI_FAMILY == WINAPI_PARTITION_DESKTOP)
+#  define WINAPI_DESKTOP
+# endif
+#endif
+
 #ifdef HAVE_SYS_MMAN_H
 # include <sys/mman.h>
 #endif
 
 #ifdef _WIN32
 # include <windows.h>
+#ifdef WINAPI_DESKTOP
 # include <wincrypt.h>
+#endif
 #else
 # include <unistd.h>
 #endif
@@ -28,10 +36,6 @@
 # define ENOSYS ENXIO
 #endif
 
-#if defined(_WIN32) && (!defined(WINAPI_FAMILY) || WINAPI_FAMILY == WINAPI_FAMILY_DESKTOP_APP)
-# define WINAPI_DESKTOP
-#endif
-
 #define CANARY_SIZE 16U
 #define GARBAGE_VALUE 0xd0
 
@@ -74,7 +78,7 @@ _sodium_memzero_as_a_weak_symbol_to_prevent_lto(void * const pnt, const size_t l
 void
 sodium_memzero(void * const pnt, const size_t len)
 {
-#ifdef _WIN32
+#ifdef WINAPI_DESKTOP
     SecureZeroMemory(pnt, len);
 #elif defined(HAVE_MEMSET_S)
     if (len > 0U && memset_s(pnt, (rsize_t) len, 0, (rsize_t) len) != 0) {
